BDT Example and readme update

justin-wong-ce opened the pull request:
Added BroadcastDistributionTable example and updated version number

ssmethurst commented on BACnetServerExample/BACnetServerExample.cpp:
```diff
@@ -566,6 +566,13 @@ int main(int argc, char** argv)
+	fpSetPropertyEnabled(g_database.device.instance, CASBACnetStackExampleConstants::OBJECT_TYPE_NETWORK_PORT, g_database.networkPort.instance, CASBACnetStackExampleConstants::PROPERTY_IDENTIFIER_BBMD_ACCEPT_FD_REGISTRATIONS, true);
+	fpSetPropertyEnabled(g_database.device.instance, CASBACnetStackExampleConstants::OBJECT_TYPE_NETWORK_PORT, g_database.networkPort.instance, CASBACnetStackExampleConstants::PROPERTY_IDENTIFIER_BBMD_BROADCAST_DISTRIBUTION_TABLE, true);
+	fpSetPropertyEnabled(g_database.device.instance, CASBACnetStackExampleConstants::OBJECT_TYPE_NETWORK_PORT, g_database.networkPort.instance, CASBACnetStackExampleConstants::PROPERTY_IDENTIFIER_BBMD_FOREIGN_DEVICE_TABLE, true);
+	uint8_t bbmdIpAddress[6] = {192, 168, 1, 208, 0xBA, 0xC0};
```
Can't used hard-coded variables in the example

justin-wong-ce commented:
Added debug callback example
Added comment to notify users about CAS BACnet Stack
Removed hardcoded BBMD Address/Port/Mask, instead asks user for input

justin-wong-ce commented:
Bumped to version 0.0.14

ssmethurst commented on BACnetServerExample/BACnetServerExample.cpp:
```diff
@@ -2075,6 +2129,16 @@ bool CallbackDeviceCommunicationControl(const uint32_t deviceInstance, const uin
 }
 
+void CallbackLogDebugMessage(const char* message, const uint16_t messageLength, const uint8_t messageType) {
+	// This callback is called when the CAS BACnet Stack logs an error or info message
```
Just dump the string to stdout in the call-back function. No need to make a global function

ssmethurst commented on BACnetServerExample/BACnetServerExample.cpp:
```diff
@@ -610,6 +657,13 @@ int main(int argc, char** argv)
 		g_database.Loop();
 
+		// Check for debug messages
+		if (debugMessage.c_str() != g_database.debugMessage && g_database.debugMessage != "") {
```
Remove this section, and just do this work in the callback. 

Also just dump the message. No need to prefix it with anything.

ssmethurst commented on CHANGELOG.md:
```diff
@@ -2,6 +2,12 @@
 ## Version 0.0.x
 
+### 0.0.14.x (2021-Jan-20)
+- Added LogDebugMessage callback to example
```
Mention the new version of the CAS BACnet Stack that has been used with the LogDebugMessage

- Updated CAS BACnet Stack to version XXXXX

ssmethurst commented on BACnetServerExample/BACnetServerExample.cpp:
```diff
@@ -566,7 +574,45 @@ int main(int argc, char** argv)
+	fpSetPropertyEnabled(g_database.device.instance, CASBACnetStackExampleConstants::OBJECT_TYPE_NETWORK_PORT, g_database.networkPort.instance, CASBACnetStackExampleConstants::PROPERTY_IDENTIFIER_BBMD_BROADCAST_DISTRIBUTION_TABLE, true);
+	fpSetPropertyEnabled(g_database.device.instance, CASBACnetStackExampleConstants::OBJECT_TYPE_NETWORK_PORT, g_database.networkPort.instance, CASBACnetStackExampleConstants::PROPERTY_IDENTIFIER_BBMD_FOREIGN_DEVICE_TABLE, true);
+	
+	// Ask for BBMD Address, Port, and Mask
```
This shouldn't be asked on start up as most people won't have a BBMD to connect to. Instead make this part of the UserInput function
